### PR TITLE
Fix CheckoutFormFragment and CheckoutConfirmationFragment. Implement Che...

### PR DESCRIPTION
**Pull request by lilshim**
Fixed Checkout form

### CODE REVIEW

**Comment by Byronium** [comment]
PHC ++

File: PHC-Android/PHC/src/main/java/phc/android/Checkout/CheckoutActivity.java [hunk 1]
```diff
@@ -25,10 +25,13 @@ protected void onCreate(Bundle savedInstanceState) {
...
         //remove this fragment.
-        transaction.add(R.id.checkout_activity_container, new CheckoutFormFragment());
+        transaction.add(R.id.checkout_activity_container, new CheckoutScannerFragment());
         transaction.commit();
         ActionBar actionbar = getActionBar();
         actionbar.setDisplayHomeAsUpEnabled(true);
     }
 
+    protected void recordResult(String code) {
+        //getServiceStatus(getKeyByValue(sOfferedServices, mServiceSelected), code);
```
Comment (kevinshieh) [inline PHC-Android/PHC/src/main/java/phc/android/Checkout/CheckoutActivity.java]: why is this here?

File: PHC-Android/PHC/src/main/java/phc/android/Checkout/CheckoutConfirmationFragment.java [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        if (savedInstanceState != null) {
+            mScanResult = savedInstanceState.getCharSequence("scan_result");
+            mManualInput = savedInstanceState.getBoolean("manual_input");
+            mExperience = savedInstanceState.getInt("experience");
+            mComments = savedInstanceState.getString("comments");
+        } else {
+            mScanResult =  getArguments().getCharSequence("scan_result");
+            mManualInput = getArguments().getBoolean("manual_input");
+            mExperience = getArguments().getInt("experience");
+            mComments =getArguments().getString("comments");
```
Comment (kevinshieh) [inline PHC-Android/PHC/src/main/java/phc/android/Checkout/CheckoutConfirmationFragment.java]: [style] space after equals

File: PHC-Android/PHC/src/main/java/phc/android/Checkout/CheckoutConfirmationFragment.java [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+     * and displays a success toast.
+     */
+    @Override
+    protected void recordScan() {
+        CheckoutActivity activity = (CheckoutActivity) getActivity();
+        activity.recordResult((String) mScanResult);
+    }
+
+
+}
```
Comment (kevinshieh) [inline PHC-Android/PHC/src/main/java/phc/android/Checkout/CheckoutConfirmationFragment.java]: [style] newline at end of file

File: PHC-Android/PHC/src/main/java/phc/android/Checkout/CheckoutConfirmationFragment.java [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+     * Records the scan result in shared preferences
+     * and displays a success toast.
+     */
+    @Override
+    protected void recordScan() {
+        CheckoutActivity activity = (CheckoutActivity) getActivity();
+        activity.recordResult((String) mScanResult);
+    }
+
+
```
Comment (kevinshieh) [inline PHC-Android/PHC/src/main/java/phc/android/Checkout/CheckoutConfirmationFragment.java]: whitespace

File: PHC-Android/PHC/src/main/java/phc/android/Checkout/CheckoutConfirmationFragment.java [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+    @Override
+    public void onSaveInstanceState(Bundle outState) {
+        super.onSaveInstanceState(outState);
+        outState.putCharSequence("scan_result", mScanResult);
+        outState.putBoolean("manual_input", mManualInput);
+        outState.putInt("experience", mExperience);
+        outState.putString("comments", mComments);
+    }
+
+
```
Comment (kevinshieh) [inline PHC-Android/PHC/src/main/java/phc/android/Checkout/CheckoutConfirmationFragment.java]: whitespace

File: PHC-Android/PHC/src/main/java/phc/android/Checkout/CheckoutConfirmationFragment.java [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+     * failure toast.
+     */
+    @Override
+    protected void retry() {
+        showFailureToast();
+        FragmentManager manager = getFragmentManager();
+        manager.popBackStack(CheckoutConfirmationFragment.TAG, FragmentManager.POP_BACK_STACK_INCLUSIVE);
+    }
+
+
```
Comment (kevinshieh) [inline PHC-Android/PHC/src/main/java/phc/android/Checkout/CheckoutConfirmationFragment.java]: whitespace

File: PHC-Android/PHC/src/main/java/phc/android/Checkout/CheckoutFormFragment.java [hunk 7]
```diff
@@ -97,19 +110,46 @@ public void onClick(View view){
...
-                    successFragment.setType(SuccessFragment.SuccessType.CHECKOUT_SUCCESS);
-                    transaction.replace(R.id.checkout_activity_container, successFragment);
-            transaction.commit();
+
+            Bundle args = new Bundle();
+            args.putCharSequence("scan_result", mScanResult);
+            args.putBoolean("manual_input", mManualInput);
+            if(mComment == null ){
+                args.putString("comments",  " ");
+            }else{
```
Comment (kevinshieh) [inline PHC-Android/PHC/src/main/java/phc/android/Checkout/CheckoutFormFragment.java]: [style] spaces before assignment operator

File: PHC-Android/PHC/src/main/java/phc/android/Checkout/CheckoutFormFragment.java [hunk 8]
```diff
@@ -97,19 +110,46 @@ public void onClick(View view){
...
+    /**
+     * Brings up another fragment when this fragment
+     * is complete
+     * @param nextFrag Fragment to display next
+     * @param fragName String fragment names
+     */
+
+    protected void displayNextFragment(Fragment nextFrag, String fragName) {
+        FragmentTransaction transaction =
+                (getActivity()).getFragmentManager().beginTransaction();
```
Comment (kevinshieh) [inline PHC-Android/PHC/src/main/java/phc/android/Checkout/CheckoutFormFragment.java]: since you call getActivity() 4 times in this class, you can set it as a class variable in an oncreate
